flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -502,9 +502,9 @@ protected void beforeInvoke() throws Exception {
 			// both the following operations are protected by the lock
 			// so that we avoid race conditions in the case that initializeState()
 			// registers a timer, that fires before the open() is called.
-			operatorChain.initializeStateAndOpenOperators(createStreamTaskStateInitializer());
```

Comment:
Good catch. I was wondering if there is any downside to putting data into buffers before initializing state. Could it be that buffers are processed before state is initialized?

Reply:
I don't think there are any issues with that, operators are kind of level above on ResultPartitions (and therefore should be initialized later).